crates/rust-analyzer/src/cargo_target_spec.rs

```diff
@@ -77,7 +77,11 @@ impl CargoTargetSpec {
             ProjectWorkspace::Cargo { cargo, .. } => {
                 let tgt = cargo.target_by_root(&path)?;
                 Some(CargoTargetSpec {
-                    package: cargo[cargo[tgt].package].name.clone(),
```

Comment:
Would `cargo[cargo[tgt].package].id.clone()` work here?

Reply:
Hmm I think it might, for now: https://docs.rs/cargo_metadata/0.9.1/cargo_metadata/struct.PackageId.html

```It is possible to inspect the repr field, if the need arises, but its precise format is an implementation detail and is subject to change.```

Would you like me to give it a try?

Reply:
Yes, and if it works, I think we need to use this field, as it **is** the package id spec. 

OTOH, it's also useful to keep this command human editable, so I think ideally the logic should be:

* if there's a single package with this name, use the name
* if there can be ambiguity, use the fully unabmigious id.

Reply:
That's a great idea, I'm going to give it a go!